chia/_tests/core/mempool/test_singleton_fast_forward.py

```diff
@@ -461,9 +461,11 @@ async def test_singleton_fast_forward_different_block(is_eligible_for_ff: bool)
             # version which is the grandchild in this scenario
             assert status == MempoolInclusionStatus.SUCCESS
             assert error is None
+            old_lineage_info = unspent_lineage_info
             unspent_lineage_info = await sim_client.service.coin_store.get_unspent_lineage_info_for_puzzle_hash(
                 singleton_puzzle_hash
             )
+            assert old_lineage_info != unspent_lineage_info
```

Comment:
These checks are redundant because we have more precise checks for `unspent_lineage_info` right below them.
```suggestion
            unspent_lineage_info = await sim_client.service.coin_store.get_unspent_lineage_info_for_puzzle_hash(
                singleton_puzzle_hash
            )
```

Reply:
is this check imprecise?

Reply:
Yep, what we want to check here is not just that our lineage info is different, we want it to be the grandchild iteration precisely, which is what we're doing right below.